bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -1145,9 +1145,10 @@ public void replicateLedgerFragment(LedgerHandle lh, final LedgerFragment ledger
         if (MapUtils.isEmpty(targetBookieAddresses)) {
             LOG.warn("Could not replicate for {} ledger: {}, not find target bookie.",
                     ledgerFragment.getReplicateType(), ledgerFragment.getLedgerId());
-            return;
+            return false;
```

Comment:
I think we can directly throw a `BKException.BKLedgerRecoveryException` here.

Reply:
+1

Reply:
Addressed. PTAL